src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -92,21 +96,36 @@ function frule((_, ΔA), ::typeof(eigen!), A::StridedMatrix{T}; kwargs...) where
 end
 
 function rrule(::typeof(eigen), A::StridedMatrix{T}; kwargs...) where {T<:Union{Real,Complex}}
-    F = eigen(A; kwargs...)
+    # NOTE: this check for hermitian-ness occurs in `eigen!`. We here do it in the rrule for
+    # `eigen` so that this works for non-mutating AD
+    isherm = ishermitian(A)
+    if isherm
+        hermA, back_Hermitian = rrule(Hermitian, A, :U)
```

Comment:
The `rrule` is only defined for `Hermitian{<:BlasReal}`. Should we also restrict `T` here to `BlasReal`? Or in a future PR?

Reply:
Outdated since I now bypass the `rrule`